gateway/sds_gateway/users/views.py

```diff
@@ -1702,6 +1746,31 @@ def _create_or_update_dataset(self, request, dataset_form) -> Dataset:
                 owner=request.user,
             )
 
+        # Persist keywords from form (comma-separated)
+        raw_keywords = dataset_form.cleaned_data.get("keywords", "") or ""
+        if raw_keywords:
+            names = [p.strip() for p in raw_keywords.split(",") if p.strip()]
+            # Deduplicate case-insensitively while preserving order
+            seen = set()
+            unique_names = []
+            for n in names:
+                key = n.lower()
+                if key in seen:
+                    continue
+                seen.add(key)
+                unique_names.append(n)
```

Comment:
1. use that slugify method here too
2. a set already has unique elements, this can be simplified